File: test/react-native/features/fixtures/scenario-launcher/src/scenarios/NativePromiseRejectionHandledScenario.js
import Scenario from './Scenario'
import Bugsnag from '@bugsnag/react-native'
import { NativeInterface } from '../lib/native'

export class NativePromiseRejectionHandledScenario extends Scenario {
  async run () {
    try {
      await NativeInterface.runScenario('NativePromiseRejectionHandledScenario')
    } catch (e) {
      Bugsnag.notify(e)
    }
  }
}

File: test/react-native/features/fixtures/scenario-launcher/android/src/main/java/com/bugsnag/reactnative/test/scenarios/NativePromiseRejectionHandledScenario.kt
package com.reactnative.scenarios

import android.content.Context
import com.bugsnag.android.Bugsnag
import com.facebook.react.bridge.Promise

class NativePromiseRejectionHandledScenario(context: Context): Scenario(context) {

    override fun run(promise: Promise) {
        promise.reject(generateException())
    }
}
